tests/expectations/test_generate_diagnostic_checklist.py

```diff
@@ -46,6 +46,82 @@ def test__convert_checks_into_output_message():
     )
 
 
+def test__count_unexpected_cases_and_get_sub_messages__with_everything_passing():
+    test_report = [
+        {
+            "test title": "positive_test_with_mostly",
+            "backend": "pandas",
+            "test_passed": "true",
+        },
+        {
+            "test title": "negative_test_with_mostly",
+            "backend": "pandas",
+            "test_passed": "true",
+        },
+        {
+            "test title": "other_negative_test_with_mostly",
+            "backend": "pandas",
+            "test_passed": "true",
+        },
+    ]
+    print(
+        ExpectColumnValuesToEqualThree()._count_unexpected_cases_and_get_sub_messages(
+            test_report
+        )
+    )
+    assert (
+        ExpectColumnValuesToEqualThree()._count_unexpected_cases_and_get_sub_messages(
+            test_report
+        )
+        == (
+            0,
+            [
+                {"message": "positive_test_with_mostly", "passed": True},
+                {"message": "negative_test_with_mostly", "passed": True},
+                {"message": "other_negative_test_with_mostly", "passed": True},
+            ],
+        )
+    )
+
+
+def test__count_unexpected_cases_and_get_sub_messages__with_one_failure():
```

Comment:
Love the tests! Are you able to add one more that shows what happens in case of test error? It might require an additional ExpectColumnValuesToEqualThree iteration.